agent/app/agent_unix.go

```diff
@@ -148,13 +149,25 @@ func isInstanceLaunchedInVPC(err error) bool {
 // a. ecs-eni
 // b. ecs-bridge
 // c. ecs-ipam
+// d. ecs-appmesh
 func (agent *ecsAgent) verifyCNIPluginsCapabilities() error {
 	// Check if we can get capabilities from each plugin
 	for _, plugin := range awsVPCCNIPlugins {
 		capabilities, err := agent.cniClient.Capabilities(plugin)
 		if err != nil {
 			return err
 		}
+		if plugin == ecscni.ECSAppMeshPluginName {
+			// check if cni plugin supports app mesh
+			if !contains(capabilities, ecscni.CapabilityAppMeshSupport) {
+				agent.cfg.AppMeshEnabled = false
```

Comment:
why is `AppMeshEnabled` part of config? 

can it be part of agent struct itself?

Reply:
this is already removed since yumeng removed capability from cni plugin